### ISSUE TITLE
Broken URL Clean-up in API Docs

### ISSUE THREAD

**Issue by yesbhautik** [mattermost/mattermost#29193]
#### Summary
 I have scanned the all URLs mentioned over API docs site. I found one broken link during scan.

```
	Line  9: ├─BROKEN─ https://forum.mattermost.org/c/dev (HTTP_404)
```

**Comment by cwarnermm** [comment]
Thanks, @yesbhautik! What tool did you use for testing the links? Are you interested in opening a mattermost PR to fix this link?

**Comment by yesbhautik** [comment]
> What tool did you use for testing the links?

@cwarnermm I used [BLC](https://github.com/stevenvachon/broken-link-checker) to scan entire Mattermost-handbook website.

With advance configuration, we also can set up auto scanning in future, using BLC or many other open-source tools are available.

**Comment by yesbhautik** [comment]
> Are you interested in opening a mattermost PR to fix this link?

Sure, which possible solution would we use?

**Comment by kasyap1234** [comment]
@cwarnermm I was thinking , this could be solved by using [https://github.com/lycheeverse/lychee-action/](url)
which is a github action , will find for broken url .

**Comment by yesbhautik** [comment]
> @cwarnermm I was thinking , this could be solved by using [https://github.com/lycheeverse/lychee-action/](url) which is a github action , will find for broken url .

Hey @kasyap1234, we can use action/scripts for automation... But before that, we have to test it properly.
Because,
- Many type of error code which tool detect, but still that URL is working fine
- Sometimes that action may face any network glitch during that exact time
- In some URL, they configured to don’t detect anything or pass different error code

Another big issue is, just finding URL is not enough, for that URL which action we will follow it is more important. Including, removing broken URL, replacement of the URL, redirection or something else.

**Comment by yesbhautik** [comment]
My suggestion is...
- Create link shortener or management server or gateway of the organization.
- At every moment it replaces the Normal link to shorted link automatically, including a UI part like comment section.
- This server will have features to handle the request. If all good, then it will be redirected to that link else show specific page.

Example: LinkedIn post, twitter post.

Yes, it is a bit complex and time-consuming method, but it is way more enterprise level professional solution, and it is most required, basic & common this day's.

Hey @cwarnermm What’s your thoughts on it?

**Comment by kasyap1234** [comment]
@yesbhautik yes I am testing it . As for fixing the broken url , replacement and redirection , 
we need to implement various scenarios after finding an invalid URL 

- If the link is completely invalid and cannot be fixed (i.e., the content no longer exists), the best approach is to remove the link.
this is what i was thinking but dont have a complete or enterprise level production ready , solution for it as of now . 
- Removing the URL: Remove or comment out links that are permanently broken.
Replacing the URL: Update the broken links with the correct or new URLs.
Handling Redirections: Follow redirects and update links accordingly.

**Comment by Joel-Wwalker** [comment]
Re-scanned the current sources under `api/`: 34 unique URLs across the yaml, md, and html. The originally reported `forum.mattermost.org/c/dev` link is no longer present in the source, and everything resolves except `https://developers.mattermost.com/api-reference` (404), which is referenced twice in `api/README.md` and once in `api/package.json`. The live page is `https://developers.mattermost.com/api-documentation/`. Fix in #37632. Happy to also add a link check scoped to `api/` in CI if there's interest in catching these automatically.
